request-manager/graph/resolver.go

```diff
@@ -442,7 +413,8 @@ func chooseConditional(n *spec.Node, jobArgs map[string]interface{}) (string, er
 	return seqName, nil
 }
 
-// Split `each: list:element` values into the list (as a slice) and the element.
+// getIterators splits `each: list:element` values into the list (as a slice)
+// and the element.
 // Returns the element name, and the slice to iterate over when repeating nodes.
 // If there is no repetition required, then it will return an empty string, "",
 // and the singleton [""], to indicate that only one iteration is needed.
```

Comment:
Maybe `Each(` is a better func name because the node spec field is called `each`. Then:
```
// Each returns list and element values for the "each:" node spec. For example, with node spec,
//
//   each:
//     - foos:foo
//     - bars:bar
//
// where jobArgs[foo]=[]string{a,b} and jobArgs[bar]=[]string{x,y}, the return values are,
//
//   []string{foo, bar}
//   [][]interface{}{
//       []interface{}{a, b}
//       []interface{}{x, y}
//   }
//
// The first return value is a list of the elements as their singular name. So with "foos:foo",
// the element name is "foo" and "foos" is the job arg providing the values in the second
// return value.
```

```

Reply:
That ^ might be wrong. This func has always been confusing to me, i.e. what list and element refer to and how the return values are supposed to be used. Would be nice to make it more intuitive, which probably requires more specific names than "list" and "element" which are very generic.